Services/Helpers/WebOSTVServiceSocketClient.m

```diff
@@ -272,14 +275,18 @@ -(void) helloTv
 
 -(void) registerWithTv
 {
-    if (self.delegate && [self.delegate respondsToSelector:@selector(socketWillRegister:)])
-        [self.delegate socketWillRegister:self];
-
     ServiceCommand *reg = [[ServiceCommand alloc] init];
     reg.delegate = self;
-
     reg.callbackComplete = ^(NSDictionary* response)
     {
+        NSString *pairingString = [response valueForKey:@"pairingType"];
+        if (pairingString) {
+            self.service.pairingType = [self pairingStringToType:pairingString];
+            if (self.delegate && [self.delegate respondsToSelector:@selector(socketWillRegister:)] && self.service.pairingType != DeviceServicePairingTypeNone){
```

Comment:
Turns out the `socketWillRegister:` name is incorrect here, because it requires the actual `pairingType` from the TV, known only after trying to register. This method should be renamed/deprecated. For now, we should leave a comment explaining its current usage.